### PR TITLE
Update to Daml Finance 1.0.0

### LINKED ISSUE #84: Trading fails with missing authorization error
To recreate in the Default.daml scenario uncomment from line 142 onwards as follows:

```
 let
    doAuction = do
      let quantity = Instrument.qty 1_000_000.0 bnd
      auctionCid        <- createAuction operator agent issuer public "AUCTION1" quantity usd 0.95 (coerceContractId ad2) issuerCashAccount
      inv1CollateralCid <- coerceContractId <$> getHolding inv1 (Instrument.qty 500_000.0 usd)
      inv2CollateralCid <- coerceContractId <$> getHolding inv2 (Instrument.qty 392_000.0 usd)
      inv3CollateralCid <- coerceContractId <$> getHolding inv3 (Instrument.qty 297_000.0 usd)
      inv1BidCid        <- createAuctionBid operator agent inv1 public auctionCid 500_000.0 1.0  inv1CollateralCid inv1SecAccount
      inv2BidCid        <- createAuctionBid operator agent inv2 public auctionCid 400_000.0 0.98 inv2CollateralCid inv2SecAccount
      inv3BidCid        <- createAuctionBid operator agent inv3 public auctionCid 300_000.0 0.99 inv3CollateralCid inv3SecAccount
      processAuction operator agent issuer public auctionCid [inv1BidCid, inv2BidCid, inv3BidCid]

    doTrade = do
      list operator exchange issuer "SWAP/USD" swp usd $ singleton public
      createAskOrder operator exchange issuer public (Id "1") (Instrument.qty 250_000.0 swp) (Instrument.qty 1.00 usd) $ singleton public
      --createAskOrder operator exchange issuer public (Id "2") (Instrument.qty 250_000.0 swp) (Instrument.qty 1.00 usd) $ singleton public
      createBidOrder operator exchange inv1  public (Id "3") (Instrument.qty 400_000.0 swp) (Instrument.qty 1.00 usd) $ singleton public
      -- createAskOrder operator exchange issuer public "1" (Instrument.qty 250_000.0 swp) (Instrument.qty 0.80 usd) $ singleton public
      -- createAskOrder operator exchange issuer public "2" (Instrument.qty 250_000.0 swp) (Instrument.qty 0.90 usd) $ singleton public
      -- createBidOrder operator exchange inv1  public "3" (Instrument.qty 500_000.0 swp) (Instrument.qty 0.90 usd) $ singleton public

  doAuction
  doTrade

  pure parties
```
Error is:
```
Default.daml",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "Script execution failed on commit at Daml.Finance.Setup.Util.Workflow:218:3:\n  22: exercise of AddObservers in Daml.Finance.Holding.Fungible:Fungible@6f8998989f6b751ba42ea1667b8893335096f2bca18ec2409d07cca6d8237edb at DA.Internal.Prelude:378:26\n      failed due to a missing authorization from 'Investor1-Default'
```
I tried adding `customer` to the `submitMulti` on line 218 of workflow.daml but this does not resolve the issue so something else clearly going on, possibly related to settlement hierarchies?

The same error can be recreated via the UI by placing a sell order then a buy order.

### PR DESCRIPTION
**Pull request by matteolimberto-da**
Fixes #84

### CODE REVIEW

**Comment by matteolimberto-da** [comment]
Testing in progress:
- Distribution (Dutch Auction) works
- Trading works
- Lifecycling is currently broken --> for some reason no instrument / effects are created upon lifecycling

File: src/daml/Daml/Finance/App/BackToBack/Service.daml [hunk 1]
```diff
@@ -36,9 +35,9 @@ template Service
       with
         id : Id
         description : Text
-        quantity : Instrument.Q
-        customerAccount : Account.K
-        providerAccount : Account.K
+        quantity : InstrumentQuantity
```
Comment (lucianojoublanc-da) [inline src/daml/Daml/Finance/App/BackToBack/Service.daml:38]: Agree with these changes. Should you also remove the type aliases `Q`, `K` from instrument? Can't see that file touched in this PR.

**Review by lucianojoublanc-da — COMMENTED** [review]
Looks good. Somebody else should also look at this; I don't feel I've enough familiarity with these changes to approve.

File: src/daml/Daml/Finance/App/BackToBack/Service.daml (hunk 1 above)
Comment (matteolimberto-da) [reply src/daml/Daml/Finance/App/BackToBack/Service.daml:38]: We should, but unfortunately those type aliases are currently used by the getting started tutorial, which is shipped as part of the Daml SDK.

I believe removing those type aliases would constitute a breaking change to an interface package. However, we have no issue tracking this item, let me raise one.

**Review by GeorgSchneider — APPROVED** [review]
I've now additionally tested (and fixed where applicable):
- Corporate Actions
- Fund Tokenization
- Securities Lending
- Structured Notes

Should be good to go.

Thanks for pulling this through

**Pull request merged**